pkg/platform/abstract/platform.go

```diff
@@ -1106,3 +1150,23 @@ func (ap *Platform) getBaseImagesOverrides() map[string]string {
 func (ap *Platform) getOnbuildImagesOverrides() map[string]string {
 	return ap.Config.ImageRegistryOverrides.OnbuildImageRegistries
 }
+
+func (ap *Platform) transformDisplayNameToName(projectConfig *platform.ProjectConfig) {
+	ap.Logger.WarnWith("Transforming display name",
+		"displayName", projectConfig.Spec.DisplayName,
+		"name", projectConfig.Meta.Name)
+
+	// trim spaces
+	displayName := strings.TrimSpace(projectConfig.Spec.DisplayName)
+
+	// lower case
+	displayName = strings.ToLower(displayName)
+
+	// no spaces
+	displayName = strings.ReplaceAll(displayName, " ", "-")
+
+	// no underscores
+	displayName = strings.ReplaceAll(displayName, "_", "-")
```

Comment:
Note that display name may have invalid stuff other than uppercase/spaces/underscores, e.g. ()!@#$%^&*, I think the transformation you did is a best effort and it's ok to stop there, and since validate called after enrich it will explode, but I do suggest adding a test for that case

Reply:
use `flect` package to handle such cases! thanks :)